tk_toolchain/cmd_line_tools/tk_build_qt_resources/__init__.py

```diff
@@ -99,28 +105,90 @@ def build_res(
     }
 
 
+def build_resources(**kwagrs):
+    build_params = {
+        "qt_ui_path": kwagrs.get("qtuipath"),
+        "py_built_path": kwagrs.get("pybuiltpath"),
+        "import_text": kwagrs.get("importtext"),
+    }
+
+    print("Building user interfaces...")
+    for index, ui_file in enumerate(kwagrs.get("uifiles")):
+        build_params_ui_files = {
+            **build_params,
+            "compiler": kwagrs.get("uic"),
+            "filename": ui_file,
+            "py_filename": (
+                kwagrs.get("uifilenames")[index] if kwagrs.get("uifilenames") else None
+            ),
+        }
+        build_qt_params = build_ui(**build_params_ui_files)
+        build_qt(**build_qt_params)
+
+    print("Building resources...")
+    for index, res_file in enumerate(kwagrs.get("resfiles")):
+        build_params_res_files = {
+            **build_params,
+            "compiler": kwagrs.get("rcc"),
+            "filename": res_file,
+            "py_filename": (
+                kwagrs.get("resfilenames")[index]
+                if kwagrs.get("resfilenames")
+                else None
+            ),
+        }
+        build_qt_params = build_res(**build_params_res_files)
+        build_qt(**build_qt_params)
+
+
+def run_yaml_commands(yaml_file, uic, rcc):
+    with open(yaml_file, "r") as file:
+        yaml = YAML()
+        yaml_commands = yaml.load(file)
+
+    for command_set in yaml_commands:
+        qt_ui_path = command_set["ui_src"]
+        py_built_path = command_set["py_dest"]
+        import_text = command_set["import_pattern"]
+        ui_files = command_set.get("ui_files", [])
+        new_names_ui_files = command_set.get("new_names_ui_files", [])
+        res_files = command_set.get("res_files", [])
+        new_names_res_files = command_set.get("new_names_res_files", [])
+
+        build_resources(
+            qtuipath=qt_ui_path,
+            pybuiltpath=py_built_path,
+            importtext=import_text,
+            uic=uic,
+            uifiles=ui_files,
+            uifilenames=new_names_ui_files,
+            rcc=rcc,
+            resfiles=res_files,
+            resfilenames=new_names_res_files,
+        )
+
+
 def main():
     parser = argparse.ArgumentParser(description="Build UI and resource files")
     parser.add_argument("-u", "--uic", help="The PySide uic compiler")
     parser.add_argument("-r", "--rcc", help="The PySide rcc compiler")
-    parser.add_argument("-p", "--pyenv", help="The Python environment path")
+    parser.add_argument(
+        "-p", "--pyenv", default=PYTHON_ENV, help="The Python environment path"
```

Comment:
Why storing this default value in a variable instead of evaluating the env var here?
```suggestion
        "-p", "--pyenv", default=os.getenv("PYTHONPATH", "/usr/bin/"), help="The Python environment path"
```

Reply:
I thought that if someone/developer has a specific 'venv' it also can be used overwriting the variable, something like this (I also let a [comment ](https://github.com/shotgunsoftware/tk-toolchain/pull/65/files#diff-9967a30257c1f6bd43330bc7d3a12ef565823225c37e4a5df0af248d0ba66381R41) about that ):
`PYTHON_ENV = "C:/Users/chaucae/Documents/projects/tk-core/.venv/Scripts"`

Reply:
Yes, indeed, it's possible. But I don't believe this answers my comment.

Reply:
Actually, I remember now. We don't want this default value. What we want is to rely on the shell's `PATH` resolution.

So the default value must be None in order to call the `uic` and `rcc` commands without a path.

Now, if we do provide a `pyenv` parameter, this will provide a way to call the `uic/rcc` commands if they can not be resolved in the `PATH`